framework/src/task_manager/taskmgr_utils.c

```diff
@@ -0,0 +1,102 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <sched.h>
+#include <errno.h>
+#include <time.h>
+#include <sys/types.h>
+#include <apps/builtin.h>
+#include <tinyara/taskmgt.h>
+#include <tinyara/clock.h>
+#include <task_manager/task_manager.h>
+#include "task_manager_internal.h"
+
+/****************************************************************************
+ * taskmgr_utils
+ ****************************************************************************/
+void tm_update_task_state(int handle)
+{
+	struct tcb_s *tcb;
+
+	tcb = sched_gettcb(TASK_PID(handle));
+	/* if terminated normally, tcb will be released, but task list is not updated yet. */
+	if (tcb == NULL && TASK_LIST_ADDR(handle) != NULL) {
+		TASK_STATUS(handle) = TM_TASK_STATE_STOP;
+	}
+}
+
+int tm_get_task_state(int handle)
+{
+	if (IS_INVALID_HANDLE(handle) || TASK_LIST_ADDR(handle) == NULL) {
+		return TM_TASK_STATE_UNREGISTERED;
+	}
+
+	return (TASK_STATUS(handle));
+}
+
+int tm_check_permission(int handle, pid_t pid)
+{
+	int ret = ERROR;
```

Comment:
I changed the default ret value to TM_FAIL_NOT_PERMITTED, but not merged in here...
Please change to TM_FAIL_NOT_PERMITTED